Some Refactoring and Addition of Configuration

Zorro909 opened the pull request.

Zorro909 committed:
Some Refactoring and Addition of Configuration

SticksDev commented on src/main/resources/RiftSystem.yml:
```diff
@@ -0,0 +1,5 @@
+ - "daily-world"
+ - "daily-world.*"
+
+portal_spawn_probability: 0.005
```
In YML, there should be a new line to tell YML that is the EOF (End Of File)

Zorro909 replied:
As far as I know, EOF is just standard EOF and there's no need to add a Line Break there...
I mean I did already test it and it's working completely fine.

Furthermore the YAML Specs don't require that at all anyway, so I'd be interested in where you got that from?

SticksDev reviewed (changes requested):
Great pull, code looks solid, you need to change some things, before we merge.

- Add 1 new line at the end of `src/main/resources/RiftSystem.yml ` Because YML requires one new line to call EOF (End Of File), or an error will be thrown when parsing.
- `.gitingore`  your `.settings` direcotry for eclipse. You can see a guide [here](https://linuxize.com/post/gitignore-ignoring-files-in-git/)

Zorro909 committed:
WIP Shape System

Portals are currently not generated as intended due to not every needed Shape being implemented

Zorro909 commented:
The yml works as it is, so I don't know why it should throw an error, but I can change that later.

Thanks for reminding me to add eclipse files to the .gitignore :)

And well, it's still Work In Progress, so there's still gonna be some things that will change before merging.

Zorro909 committed:
Add Eclipse .gitignore Entries

Thatcooldevguy commented:
Huh, okay. Github was complaining

Zorro909 committed:
Added FilterTransformations and Sphere

Zorro909 committed:
ShapeSystem 1.0 finished

Added shape native BlockTransformers

Finished Refactoring of RiftHunters to accomodate changed StructureSystem

pyxld-kris commented:
This is sooo much prettier, thank you! I also REALLY appreciate the addition of 'allowedWorlds' in the config. That's going to let us ship this way sooner 🥇

pyxld-kris reviewed (approved):
🔥
